src/main/java/gomedic/model/person/UniqueAbstractPersonList.java

```diff
@@ -29,6 +29,21 @@
     private final ObservableList<T> internalUnmodifiableList =
             FXCollections.unmodifiableObservableList(internalList);
 
+    /**
+     * Returns a new available id.
+     * If it's an empty list, return id 1.
+     */
+    public int getNewId() {
+        try {
+            return internalList
+                    .get(internalList.size() - 1)
+                    .getId()
+                    .getIdNumber() + 1;
+        } catch (Exception e) {
+            return 1;
+        }
+    }
```

Comment:
Maybe we should change this according to your suggestion
```suggestion
    public int getNewId() {
        try {
            return internalList
                    .get(internalList.size() - 1)
                    .getId()
                    .getIdNumber() + 1;
        } catch (Exception e) {
            return 1;
        }
    }
    public int getNewId() {
	        if (internalList.size() == 0) {
	            return 1;
	        }
	
	        int counter = 1;
	        // Find first id that does not exist in the list starting from D001
	        for (Iterator<Integer> it = internalList
	                .stream()
	                .map(entry -> entry.getActivityId().getIdNumber())
	                .sorted()
	                .iterator(); it.hasNext(); counter++) {
	            int id = it.next();
	            if (id != counter) {
	                return counter;
	            }
	        }
	
	        // all the id are in sequences, make a new one
	        if (counter <= MAX_CAPACITY) {
	            return counter;
	        }
	
	        throw new MaxAddressBookCapacityReached();
	    }
```

Reply:
Sorry i didnt realize that u were working on the id part as well, I added my own version of it too for the doctor class, lemme know if i shld change it